pkg/operator/edgefs/iscsi/controller.go

```diff
@@ -159,6 +163,31 @@ func (c *ISCSIController) serviceOwners(service *edgefsv1beta1.ISCSI) []metav1.O
 	return []metav1.OwnerReference{c.ownerRef}
 }
 
+func (c *ISCSIController) ParentClusterChanged(cluster edgefsv1beta1.ClusterSpec) {
+	if c.rookImage == cluster.EdgefsImageName {
+		logger.Infof("No need to update the iscsi service, the same images present")
+		return
+	}
+
+	// update controller options by updated cluster spec
+	c.rookImage = cluster.EdgefsImageName
```

Comment:
You might need to lock this method with a mutex. The parent controller would be running in a different goroutine from the child controllers, so it is possible the child controller will be busy with some operation when this callback comes from the parent controller. But maybe you'll be fine if you're just basically updating k8s resources.